packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -411,6 +414,29 @@ export const DatePickerMixin = (subclass) =>
       );
 
       this.addController(new VirtualKeyboardController(this));
+
+      this.$.overlay.addEventListener('opened-changed', (e) => {
+        this.opened = e.detail.value;
+      });
+
+      this.$.overlay.addEventListener('vaadin-overlay-escape-press', () => {
+        this._focusedDate = this._selectedDate;
+        this._close();
+      });
+
+      this.addEventListener('mousedown', () => this.__bringToFront());
+      this.addEventListener('touchstart', () => this.__bringToFront());
+
+      this.$.overlay.renderer = (root) => {
```

Comment:
**nit:** How about we make the whole renderer a method instead of having `_initOverlayContent` that only does a partial job?

Reply:
Good point, done.